superflore/generators/bitbake/run.py

```diff
@@ -76,6 +77,22 @@ def main():
         warn('"{0}" distro detected...'.format(args.ros_distro))
         selected_targets = [args.ros_distro]
         preserve_existing = False
+    # open cached tar file if it exists
+    md5_cache = None
+    sha256_cache = None
+    if args.tar_dir:
+        try:
+            md5_file = open('%s/md5_cache.pickle', 'rb')
+            md5_cache = pickle.load(md5_file)
+            md5_file.close()
+        except IOError:
+            md5_cache = dict()
+        try:
+            sha256_file = open('%s/md5_cache.pickle', 'rb')
+            sha256_cache = pickle.load(sha256_file)
```

Comment:
Are there matching write/dump calls to update the cache with the new hashes somewhere?

Reply:
Not yet -- this is still a WIP. Still deciding where to out those.